presto-main/src/test/java/io/prestosql/sql/query/TestSubqueries.java

```diff
@@ -64,14 +53,37 @@ public void teardown()
     @Test
     public void testCorrelatedExistsSubqueriesWithOrPredicateAndNull()
     {
-        assertExistsRewrittenToAggregationAboveJoin(
+        assertions.assertQueryAndPlan(
                 "SELECT EXISTS(SELECT 1 FROM (VALUES null, 10) t(x) WHERE y > x OR y + 10 > x) FROM (values 11 + if(rand() >= 0, 0)) t2(y)",
                 "VALUES true",
-                false);
-        assertExistsRewrittenToAggregationAboveJoin(
+                anyTree(
+                        aggregation(
+                                ImmutableMap.of("COUNT", functionCall("count", ImmutableList.of("NON_NULL"))),
+                                aggregation -> aggregation.isStreamable() && aggregation.getStep() == SINGLE,
+                                node(JoinNode.class,
+                                        anyTree(
+                                                node(ValuesNode.class)),
```

Comment:
When refactoring this, you could replace all `node(ValuesNode.class))` with `values("x")`, `values("y")` etc, accordingly to the actual names used in queries. It would make it easier to follow how the plan has transformed.